pdns/dnsdist.cc

```diff
@@ -692,11 +692,23 @@ std::shared_ptr<ServerPool> createPoolIfNotExists(pools_t& pools, const string&
     if (!poolName.empty())
       vinfolog("Creating pool %s", poolName);
     pool = std::make_shared<ServerPool>();
+    pool->policy = NULL;
```

Comment:
Just make sure it's default initialized to `nullptr` by replacing:
`std::shared_ptr<ServerPolicy> policy;`
by
`std::shared_ptr<ServerPolicy> policy{nullptr};`